armi/nucDirectory/thermalScattering.py

```diff
@@ -195,64 +189,78 @@ def _genACELabel(self):
             label = f"{element.symbol.lower()}-{first.a:d}"
         else:
             raise ValueError(f"{self} label cannot be generated")
+
         return label
 
 
-def factory():
+def factory(byName):
     """
     Generate the :class:`ThermalScattering` instances.
 
-    The logic for these is a bit complex so we skip reading a text file and code
-    it up here.
+    The logic for these is a bit complex so we skip reading a text file and code it up here.
 
-    This is called by the nuclideBases factory since it must ALWAYS be re-run when
-    the nuclideBases are rebuilt.
+    This is called by the nuclideBases factory since it must ALWAYS be re-run when the nuclideBases are rebuilt.
 
     See Also
     --------
     armi.nucDirectory.nuclideBases.factory
         Calls this during ARMI initialization.
-
-    .. warning::
-        This gets called automatically during init, so don't call it
-        unless you know what you're doing.
     """
     global byNbAndCompound
     byNbAndCompound.clear()
 
-    al27 = nb.byName["AL27"]
-    fe56 = nb.byName["FE56"]
-    be = nb.byName["BE"]
-    c = nb.byName["C"]
-    d = nb.byName["H2"]
-    n = nb.byName["N"]
-    o = nb.byName["O"]
-    h = nb.byName["H"]
-    u = nb.byName["U"]
-    zr = nb.byName["ZR"]
-    si = nb.byName["SI"]
-
-    for isotope in [al27, fe56]:
-        byNbAndCompound[isotope, None] = ThermalScattering(isotope)
-
-    byNbAndCompound[be, BE_METAL] = ThermalScattering(
-        be, BE_METAL, endf8Label=f"tsl-{BE_METAL}.endf", aceLabel="be-met"
-    )
-    byNbAndCompound[be, BEO] = ThermalScattering(be, BEO, endf8Label=BEO, aceLabel="be-beo")
-    byNbAndCompound[c, SIC] = ThermalScattering(c, SIC)
-    byNbAndCompound[d, D2O] = ThermalScattering(d, D2O, f"tsl-Din{D2O}.endf", "d-d2o")
-    byNbAndCompound[h, H2O] = ThermalScattering(h, H2O)
-    byNbAndCompound[h, ZRH] = ThermalScattering(h, ZRH)
-    byNbAndCompound[n, UN] = ThermalScattering(n, UN)
-    byNbAndCompound[o, BEO] = ThermalScattering(o, BEO)
-    byNbAndCompound[o, D2O] = ThermalScattering(o, D2O, f"tsl-Oin{D2O}.endf", "o-d2o")
-    byNbAndCompound[o, UO2] = ThermalScattering(o, UO2)
-    byNbAndCompound[u, UO2] = ThermalScattering(u, UO2)
-    byNbAndCompound[u, UN] = ThermalScattering(u, UN)
-    byNbAndCompound[zr, ZRH] = ThermalScattering(zr, ZRH)
-    byNbAndCompound[si, SIC] = ThermalScattering(si, SIC)
-    byNbAndCompound[c, CRYSTALLINE_GRAPHITE] = ThermalScattering(
-        c, CRYSTALLINE_GRAPHITE, f"tsl-{CRYSTALLINE_GRAPHITE}.endf", "grph"
-    )
-    byNbAndCompound[c, GRAPHITE_10P] = ThermalScattering(c, GRAPHITE_10P, f"tsl-{GRAPHITE_10P}.endf", "grph10")
-    byNbAndCompound[c, GRAPHITE_30P] = ThermalScattering(c, GRAPHITE_30P, f"tsl-{GRAPHITE_30P}.endf", "grph30")
+    if "AL27" in byName:
+        al27 = byName["AL27"]
```

Comment:
Controversial take, no action needed.

---

This is where the walrus operator would be excellent!
```suggestion
    if (al27 := byName.get("AL27")) is not None:
```
It's been around since py38 so it'll be supported in released ARMI.